core/domain/user_services.py

```diff
@@ -160,11 +183,17 @@ def get_multi_user_ids_from_usernames(
     username_to_user_id_map = {
         model.normalized_username: model.id for model in found_models
     }
+    user_ids = []
+    for username in normalized_usernames:
+        user_id = username_to_user_id_map.get(username)
+        if strict:
```

Comment:
Optional: could simplify to "if strict and user_id is None"

Reply:
Done